modules/oe_content_call_proposals/src/CallForProposalsNodeWrapper.php

```diff
@@ -0,0 +1,205 @@
+    $opening_date->setTime(0, 0, 0);
+    return $opening_date;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDeadlineDate(): ?DrupalDateTime {
+    if (!$this->hasDeadlineDate()) {
+      return NULL;
+    }
+
+    $dates = $this->entity->get('oe_call_proposals_deadline')->getValue();
+
+    if (!is_array($dates)) {
+      return NULL;
+    }
+
+    $values_column = array_column($dates, 'value');
+    // Get the latest date in relation to the "closed" proposal's status.
+    array_multisort($values_column, SORT_DESC, $dates);
+
+    $date = current($dates);
+    return new DrupalDateTime($date['value']);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getStatus(): string {
+    $opening_date = $this->getOpeningDate();
+    $closing_date = $this->getDeadlineDate();
+    $now = $this->getNow();
+
+    $status = static::STATUS_NOT_AVAILABLE;
+    if (isset($opening_date)) {
+      if ($now < $opening_date) {
+        $status = static::STATUS_UPCOMING;
+      }
+      else {
+        $status = static::STATUS_OPEN;
+      }
+    }
+    if (isset($closing_date) && $now > $closing_date) {
+      $status = static::STATUS_CLOSED;
+    }
+
+    return $status;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getStatusLabel(): MarkupInterface {
+    switch ($this->getStatus()) {
+      case static::STATUS_UPCOMING:
+        $label = $this->t('Upcoming');
+        break;
+
+      case static::STATUS_OPEN:
+        $label = $this->t('Open');
+        break;
+
+      case static::STATUS_CLOSED:
+        $label = $this->t('Closed');
+        break;
+
+      default:
+        $label = $this->t('N/A');
+    }
+
+    return $label;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function hasStatus(): bool {
+    return $this->getStatus() !== static::STATUS_NOT_AVAILABLE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isUpcoming(): bool {
+    return $this->getStatus() === static::STATUS_UPCOMING;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isOpen(): bool {
+    return $this->getStatus() === static::STATUS_OPEN;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isClosed(): bool {
+    return $this->getStatus() === static::STATUS_CLOSED;
+  }
+
+  /**
+   * Gets current time.
+   *
+   * @return \Drupal\Core\Datetime\DrupalDateTime
+   *   Current time.
+   */
+  public function getNow(): DrupalDateTime {
+    $request_time = \Drupal::service('datetime.time')->getRequestTime();
+    return DrupalDateTime::createFromTimestamp($request_time);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getModelsList(): array {
+    return [
+      CallForProposalsNodeWrapperInterface::MODEL_SINGLE_STAGE => t('Single-stage'),
+      CallForProposalsNodeWrapperInterface::MODEL_TWO_STAGE => t('Two-stage'),
+      CallForProposalsNodeWrapperInterface::MODEL_MULTIPLE_CUT_OFF => t('Multiple cut-off'),
+      CallForProposalsNodeWrapperInterface::MODEL_PERMANENT => t('Permanent'),
```

Comment:
Use `$this->t('...');`

Reply:
The error `Using $this when not in object context` is triggered when I use `$this->t()`.

Reply:
`t()` should be used in public static function.

Reply:
Sorry my bad!